src/api/c/imageio2.cpp

```diff
@@ -61,9 +61,9 @@ static af_err readImage_t(af_array *rImage, const uchar* pSrcLine, const int nSr
                 } else {
                     // Non 8-bit types do not use ordering
                     // See Pixel Access Functions Chapter in FreeImage Doc
-                    pDst0[indx] = (T) *(src + (x * step + 0));
-                    pDst1[indx] = (T) *(src + (x * step + 1));
-                    pDst2[indx] = (T) *(src + (x * step + 2));
+                    pDst0[indx] = (T) *(src + (x * step + FI_RGBA_RED));
+                    pDst1[indx] = (T) *(src + (x * step + FI_RGBA_GREEN));
+                    pDst2[indx] = (T) *(src + (x * step + FI_RGBA_BLUE));
```

Comment:
Yes, but I changed everything to use macros in imageio. I am removing the macros for non 8 bit types.